.github/workflows/build_and_test.yml

```diff
@@ -78,7 +78,7 @@ jobs:
     - name: upload ACE_TAO artifact
       uses: actions/upload-artifact@v2
       with:
-        name: ubuntu_ace_artifact
+        name: ubuntu-20_04_defaults_ace_artifact
```

Comment:
Will it not work when you use the job information, something like {{ job.name }}, would make it much easier for maintenance. You can also refer to the properties another job, see https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idname.